llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -936,13 +937,83 @@ Error MCCASBuilder::createStringSection(
   return Error::success();
 }
 
+/// Reads and returns the length field of a debug info header contained in
+/// Reader, assuming Reader is positioned at the beginning of the section.
+/// The Reader's state is advanced to the first byte after the section.
+Expected<size_t> getSizeFromDebugHeaderAndSkip(BinaryStreamReader &Reader) {
+  // From Dwarf 5 section 7.4:
+  // In the 32-bit DWARF format, an initial length field [...] is an unsigned
+  // 4-byte integer (which must be less than 0xfffffff0);
+  uint32_t Word1;
+  if (auto E = Reader.readInteger(Word1))
+    return E;
+
+  // TODO: handle 64-bit DWARF format.
+  assert(Word1 < 0xfffffff0);
```

Comment:
It would be much better to return an error here, since we are already returning `Expected<>`. The assertions are going to be stripped from release builds and then we just opened the code up to some hard to debug crashes.

Reply:
I second this, I understand why the assert is here but why not just return an Error instead?

Also correct me if I am wrong, but I thought if the value is 0xffffffff it is considered 64-bit and everything else is considered 32-bit DWARF, so why check for `< 0xfffffff0`?

Reply:
Will change the assertion to an error!

> correct me if I am wrong, but I thought if the value is 0xffffffff it is considered 64-bit

This assumption is incorrect, see the comment on line 946 for a direct quote from the spec. In short:
1.  it is the 32-bit format if the value is `< 0xfffffff0`
2. It is the 64-bit format if the value is `== 0xffffffff`
3. It is neither otherwise

Reply:
That's correct.